tests/cases/data/source/MongoDbTest.php

```diff
@@ -630,125 +537,18 @@ public function testRelationshipGenerationWithPluralNamingConvention() {
-		});
-	}
-
-	public function testUpdateNoConnectionException() {
-		$db = new MongoDb(['host' => '__invalid__', 'autoConnect' => false]);
-		$this->assertException('Could not connect to the database.', function() use ($db) {
-			$db->update(null);
-		});
-	}
-
-	public function testDeleteNoConnectionException() {
-		$db = new MongoDb(['host' => '__invalid__', 'autoConnect' => false]);
-		$this->assertException('Could not connect to the database.', function() use ($db) {
-			$db->delete(null);
-		});
-	}
-
-	public function testSourcesNoConnectionException() {
-		$db = new MongoDb(['host' => null, 'autoConnect' => false]);
-		$this->assertException('Could not connect to the database.', function() use ($db) {
-			$db->sources(null);
-		});
-	}
-
-	public function testAtomicUpdate() {
-		MockMongoPost::config(['meta' => ['source' => 'posts']]);
-		$data = ['initial' => 'one', 'values' => 'two'];
-
-		$this->_db->connection = new MockMongoConnection();
-		$this->_db->connection->results = [true, true];
-
-		$document = MockMongoPost::create($data);
-		$this->assertTrue($document->save());
-
-		$result = array_shift($this->_db->connection->queries);
-		$expected = [
-			'type' => 'insert',
-			'collection' => 'posts',
-			'data' => ['initial' => 'one', 'values' => 'two', '_id' => $document->_id],
-			'options' => [
-				'validate' => true, 'events' => 'create', 'whitelist' => null, 'callbacks' => true,
-				'locked' => false, 'w' => 1, 'wTimeoutMS' => 10000, 'fsync' => false
-			]
-		];
-		$this->assertEqual($expected, $result);
-
-		$duplicate = MockMongoPost::create(['_id' => $document->_id], ['exists' => true]);
-		$duplicate->values = 'new';
-		$this->assertTrue($duplicate->save());
-
-		$result = array_shift($this->_db->connection->queries);
-		$expected = [
-			'type' => 'update',
-			'collection' => 'posts',
-			'conditions' => ['_id' => $document->_id],
-			'update' => ['$set' => ['values' => 'new']],
-			'options' => [
-				'validate' => true, 'events' => 'update', 'whitelist' => null,
-				'callbacks' => true, 'locked' => false, 'upsert' => false, 'multiple' => true,
-				'w' => 1, 'wTimeoutMS' => 10000, 'fsync' => false
-			]
-		];
-		$this->assertEqual($expected, $result);
-
-		array_push($this->_db->connection->results, new MockResultResource(['data' => [
-			['_id' => $duplicate->_id, 'initial' => 'one', 'values' => 'new']
-		]]));
-
-		$document = MockMongoPost::find($duplicate->_id);
-		$expected = ['_id' => (string) $duplicate->_id, 'initial' => 'one', 'values' => 'new'];
-		$this->assertEqual($expected, $document->data());
-
-		$result = array_shift($this->_db->connection->queries);
-		$expected = [
-			'type' => 'find', 'collection' => 'posts', 'fields' => [], 'conditions' => [
-				'_id' => $duplicate->_id
-			]
-		];
-		$this->assertEqual($expected, $result);
-	}
-
-	/**
-	 * Tests that the MongoDB adapter will not attempt to overwrite the _id field on document
-	 * update.
-	 */
-	public function testPreserveId() {
```

Comment:
Looks like a valuable test to me, are we checking for this case implicitly elsewhere now?